### PR TITLE
[Docs] Add contents to the docs

### PR DESCRIPTION
**Pull request by pimvenderbosch**
In this PR the contents of the documentation pages will be filled.

- [x] Improve docstrings
- [x] Write general documentation pages

### CODE REVIEW

**Comment by pimvenderbosch** [comment]
Hi @RolandMacDoland and @Doomsk,

I've completed half of this PR, namely the updating of docstrings. Possibly a good idea to already review this part, so that the eventual review is not massive.

Best,
Pim

File: qadence2_ir/irast.py [hunk 1]
```diff
@@ -9,24 +17,24 @@
 
 
 class AST:
```
[the remark this answers is not in the document]
Comment (pimvenderbosch) [reply qadence2_ir/irast.py:19]: The `IRBuilder.parse_sequence` implementation in Qadence 2 expressions translates an expression tree into an `AST` object, see [here](https://github.com/pasqal-io/qadence2-expressions/blob/3fb112cf684c8dab28fb4535e8cb1355189d4be6/qadence2_expressions/ircompiler.py#L60). Other front-ends need to provide their own implementation.

**Review by Doomsk — CHANGES_REQUESTED** [review]
Thanks for the great work. There are a few comments to improve the documentation

File: docs/examples.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
+## Examples
```
Comment (Doomsk) [inline docs/examples.md:1]: This one is not being used or referenced in the documentation. Is it expected, i.e. to be used later somewhere in the documentation?

**Review by Doomsk — CHANGES_REQUESTED** [review]
Great work! A few comments and I think that's it

File: docs/examples.md (hunk 2 above)
Comment (pimvenderbosch) [reply docs/examples.md:1]: Indeed, I'm still working on the examples page right now.

**Review by Doomsk — APPROVED** [review]

**Comment by pimvenderbosch** [comment]
Created #22 with ideas of improving the documentation at a later stage.

**Pull request merged**